kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/FilterContext.java

```diff
@@ -144,4 +145,41 @@ <M extends ApiMessage> CompletionStage<M> sendRequest(RequestHeaderData header,
      */
     Optional<ClientTlsContext> clientTlsContext();
 
+    /**
+     * Allows a filter (typically one which implements {@link SaslAuthenticateRequestFilter})
+     * to announce a successful authentication outcome with the Kafka client to other plugins.
+     * After calling this method the result of {@link #clientSaslContext()} will
+     * be non-empty for this and other filters.
+     *
+     * In order to support reauthentication, calls to this method and
+     * {@link #clientSaslAuthenticationFailure(String, String, Exception)}
+     * may be arbitrarily interleaved during the lifetime of a given filter instance.
+     * @param mechanism The SASL mechanism used
+     * @param authorizedId The authorizedId
+     */
+    void clientSaslAuthenticationSuccess(String mechanism,
+                                         String authorizedId);
+
+    /**
+     * Allows a filter (typically one which implements {@link SaslAuthenticateRequestFilter})
+     * to announce a failed authentication outcome with the Kafka client.
+     * It is the filter's responsilbity to return the right error response to a client, and/or disconnect.
```

Comment:
The other method describes the effect it has on `#clientSaslContext()`, this one should do the same, talking about how after calling this method, the result will be empty.